GreenOpsInc/argo-workflows
# EM-104: Update few lines of Argo Workflows UI to allow proper embedding with GreenOps

## Issue

Update Argo Workflows server in order to restrict iframe embedding on all websites but Greenops UI.
Also, change cookies generation so Greenops UI will be able to access them.

## Solution

- [x] updated cookie handling for sso and client auth
- [x] updated CSP header to allow iframe embedding only on Greenops website

## Testing

Currently, we setting a CSP policy to allow requests from https://localhost:8081/. In future, this should be changed to the Greenops UI production URL.

Steps to build Greenops version of argo-server:

1. build ui:
    
    ```bash
    export STATIC_FILES=true
    make cli
    ```
    
2. build argo-server image:
    
    ```bash
    docker build \
        --platform="linux/amd64" \
        --target argocli \
        --tag greenopsinc/argo-server:3.3.8 .
    ```
    
3. load the image into minikube:
    
    ```bash
    minikube image load greenopsinc/argo-server:3.3.8
    ```
    
4. deploy updated argo-server deployment:
    
    ```bash
    kubectl apply -f ./docs/greenops/argo-server-deployment-example.yaml -n argo
    ```
    

Try to login on the argo workflows side and open pipeline run details page on greenops UI. Iframe should be visible.

Try to embed iframe to some other website. You should see an error in the iframe:
![image](https://user-images.githubusercontent.com/9743942/173084553-a34b5571-539b-4332-ba0a-180ec565d1e3.png)
